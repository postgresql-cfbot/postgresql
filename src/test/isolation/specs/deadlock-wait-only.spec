setup
{
  CREATE TABLE a1 ();
}

teardown
{
  DROP TABLE a1;
}

session s1
setup		{ BEGIN; }
step s1re	  { LOCK TABLE a1 IN ROW EXCLUSIVE MODE; }
step s1aewo	{ LOCK TABLE a1 IN ACCESS EXCLUSIVE MODE WAIT ONLY; }
step s1c	  { COMMIT; }

session s2
setup		{ BEGIN; }
step s2as		{ LOCK TABLE a1 IN ACCESS SHARE MODE; }
step s2swo	{ LOCK TABLE a1 IN SHARE MODE WAIT ONLY; }
step s2c		{ COMMIT; }

permutation s1re s2as s2swo s1aewo s1c s2c
