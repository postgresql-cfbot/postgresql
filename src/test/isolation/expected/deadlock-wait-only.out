Parsed test spec with 2 sessions

starting permutation: s1re s2as s2swo s1aewo s1c s2c
step s1re: LOCK TABLE a1 IN ROW EXCLUSIVE MODE;
step s2as: LOCK TABLE a1 IN ACCESS SHARE MODE;
step s2swo: LOCK TABLE a1 IN SHARE MODE WAIT ONLY; <waiting ...>
step s1aewo: LOCK TABLE a1 IN ACCESS EXCLUSIVE MODE WAIT ONLY; <waiting ...>
step s1aewo: <... completed>
step s2swo: <... completed>
ERROR:  deadlock detected
step s1c: COMMIT;
step s2c: COMMIT;
